modules/extension/wmts/src/main/java/org/geotools/ows/wmts/response/WMTSGetCapabilitiesResponse.java

```diff
@@ -50,47 +48,51 @@ public class WMTSGetCapabilitiesResponse extends GetCapabilitiesResponse {
     private static final Logger LOGGER = Logging.getLogger(WMTSGetCapabilitiesResponse.class);
     private static WMTSConfiguration WMTS_CONFIGURATION = new WMTSConfiguration();
 
+    private static final int MAX_BUFFER_SIZE = 2048; // Should be enough to support the size of a ServiceException.
```

Comment:
I would have merged but the PR is still marked as draft.
While I'm here, 2048 is probably enough but you never know, maybe add  a system variable for this limit so that one can tweak it, just in case.